.github/workflows/overcloud-host-image-build.yml

```diff
@@ -98,148 +126,242 @@ jobs:
+        continue-on-error: true
         run: |
           source venvs/kayobe/bin/activate &&
           source src/kayobe-config/kayobe-env --environment ci-builder &&
           kayobe overcloud host image build --force-rebuild \
-          -e os_distribution=ubuntu \
-          -e os_release=jammy \
+          -e os_distribution="ubuntu" \
+          -e os_release="jammy" \
           -e stackhpc_overcloud_dib_name=overcloud-ubuntu-jammy
         env:
           KAYOBE_VAULT_PASSWORD: ${{ secrets.KAYOBE_VAULT_PASSWORD }}
         if: inputs.ubuntu-jammy
 
-      - name: Upload Ubuntu Jammy 22.04 overcloud host image artifact
+      - name: Upload Ubuntu 22.04 host image to ARK
         run: |
           source venvs/kayobe/bin/activate &&
           source src/kayobe-config/kayobe-env --environment ci-builder &&
           kayobe playbook run \
           src/kayobe-config/etc/kayobe/ansible/pulp-host-image-upload.yml \
-          -e image_path='/opt/kayobe/images/overcloud-ubuntu-jammy' \
-          -e os_distribution='ubuntu' \
-          -e os_release=jammy
+          -e image_path=/opt/kayobe/images/overcloud-ubuntu-jammy \
+          -e host_image_tag=${{ steps.host_image_tag.outputs.host_image_tag }} \
+          -e os_distribution="ubuntu" \
+          -e os_release="jammy"
         env:
           KAYOBE_VAULT_PASSWORD: ${{ secrets.KAYOBE_VAULT_PASSWORD }}
-        if: inputs.ubuntu-jammy
+        if: inputs.ubuntu-jammy && steps.build_ubuntu_jammy.outcome == 'success'
 
-      - name: Upload updated images artifact
-        uses: actions/upload-artifact@v3
-        with:
-          name: Updated images list
-          path: /tmp/updated_images.txt
-          retention-days: 7
-        if: always()
+      - name: Upload Ubuntu 22.04 host image to SMS
+        run: |
+          source venvs/kayobe/bin/activate &&
+          openstack image create \
+          overcloud-ubuntu-jammy-${{ steps.host_image_tag.outputs.host_image_tag }} \
+          --container-format bare \
+          --disk-format qcow2 \
+          --file /opt/kayobe/images/overcloud-ubuntu-jammy/overcloud-ubuntu-jammy.qcow2 \
+          --private \
+          --os-cloud sms-lab-release \
+          --progress
+        env:
+          OS_APPLICATION_CREDENTIAL_ID: ${{ secrets.OS_APPLICATION_CREDENTIAL_ID }}
+          OS_APPLICATION_CREDENTIAL_SECRET: ${{ secrets.OS_APPLICATION_CREDENTIAL_SECRET }}
+        if: inputs.ubuntu-jammy && steps.build_ubuntu_jammy.outcome == 'success'
 
-      - name: Clean up old images
+      - name: Fail if any host image builds failed
         run: |
-          sudo rm -rf /opt/kayobe/images/
-        if: always()
+          exit 1
+        if: steps.build_centos_stream_8.outcome == 'failure' ||
+            steps.build_rocky_8.outcome == 'failure' ||
+            steps.build_rocky_9.outcome == 'failure' ||
+            steps.build_ubuntu_focal.outcome == 'failure' ||
+            steps.build_ubuntu_jammy.outcome == 'failure'
```

Comment:
While I can see that something failed, it's not currently that clear what it was: https://github.com/stackhpc/stackhpc-kayobe-config/actions/runs/4981847393/jobs/8916671378